### PR TITLE
Add additional error metadata.

### PR DESCRIPTION
**Pull request by nowells**
It would sometimes be useful to have the rich metadata about a given command provided for debugging.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -32,22 +32,29 @@ function spawn(command, args, options, done) {
     child.on('error', onerror)
     child.on('close', onclose)
 
+    function fixupError(error) {
+      error.stdout = out
+      error.stderr = err
+      error.command = command;
```
Comment (NickHeiner) [inline lib/index.js]: Can you be consistent with the semicolons or lack thereof?

File: lib/index.js [hunk 2]
```diff
@@ -32,22 +32,29 @@ function spawn(command, args, options, done) {
     child.on('error', onerror)
     child.on('close', onclose)
 
+    function fixupError(error) {
```
Comment (NickHeiner) [inline lib/index.js]: nit: `fixUpError`, since "fix" and "up" are different words. 😄

File: lib/index.js (hunk 1 above)
Comment (nowells) [inline lib/index.js]: will do!

**Comment by nowells** [comment]
@NickHeiner should be cleaned up now.

**Comment by NickHeiner** [comment]
:shipit:

File: lib/index.js [hunk 3]
```diff
@@ -32,22 +32,29 @@ function spawn(command, args, options, done) {
     child.on('error', onerror)
     child.on('close', onclose)
 
+    function fixUpError(error) {
```
Comment (NickHeiner) [inline lib/index.js]: 🎱

**Review by NickHeiner — APPROVED** [review]

File: lib/index.js (hunk 3 above)
Comment (NickHeiner) [inline lib/index.js]: Amusingly, there does not seem to be a way to do the system-recognized "approved PR" without adding a comment on a line.

**Comment by nowells** [comment]
Thanks for the review @NickHeiner!

**Comment by nowells** [comment]
@NickHeiner let me know if you need anything else from me before you merge?

**Comment by NickHeiner** [comment]
I don't need anything else –  I assumed that you would do it?

**Comment by NickHeiner** [comment]
Are you able to publish to npm or shall I do it?

**Comment by nowells** [comment]
@NickHeiner I don't think I have access (not an admin on this project on github or npm). Thanks!!!

**Comment by nowells** [comment]
@NickHeiner looks like I have `npm publish` access so I published, just don't have github access. Thanks for the review!

**Comment by haoxins** [comment]
@nowells added you as a collaborator!

**Comment by nowells** [comment]
> @nowells added you as a collaborator!

Thanks @coderhaoxin!

**Pull request merged**